src/rf/apps/uploads/views.py

```diff
@@ -3,31 +3,50 @@
 from __future__ import unicode_literals
 from __future__ import division
 
-# NEEDED WHEN WE ADD PROPER AUTHORIZATION
-# from django.contrib.auth.models import User
+from django.contrib.auth.models import User
 
-from rest_framework.response import Response
-from rest_framework import decorators
-from rest_framework.permissions import AllowAny
+from rest_framework import status
+from rest_framework.exceptions import APIException
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.renderers import StaticHTMLRenderer
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from django.conf import settings
 
 import boto
 import boto.auth
 import boto.provider
+import uuid
+
+
+class SignUploadRequestView(APIView):
+    permission_classes = [IsAuthenticated]
+    renderer_classes = (StaticHTMLRenderer, )
+
+    # TODO - WE'LL WANT TO ENSURE THE USER CAN IN FACT ADD NEW CONTENT TO THE
+    # BUCKET BEFORE WE SIGN THE REQUEST.
+    def get(self, request, format=None):
+        user_id = request.user.id
+        bucket_name = settings.AWS_BUCKET_NAME + '/'
+        to_sign = str(request.REQUEST.get('to_sign'))
+
+        # Extract file name from the string to sign and then extract the user
+        # id from that.
+        start = len(bucket_name) + to_sign.find(bucket_name)
+        end = to_sign.rfind('?uploads', start)
+        file_name = to_sign[start:end]
+        sent_user_id = int(file_name[0:file_name.find('-')])
```

Comment:
Consider removing for these reasons:
1. Quotas will be enforced against the logged-in user.
2. UUID should make it nearly impossible to override an existing image on S3.
3. After geoprocessing completes, the images on S3 are tiled and then deleted? So even successful upload request forgeries would be ineffectual.